Windows JPEG builds are failing on CI

## 🐛 Bug

CI today has been failing with the following error
```
C:\tools\MINICO~1\CONDA-~2\TORCHV~1\work\torchvision\csrc\io\image\cpu\encode_jpeg.cpp(77): error C2664: 'void jpeg_mem_dest(j_compress_ptr,unsigned char **,size_t *)': cannot convert argument 3 from 'unsigned long *' to 'size_t *'
```

See https://app.circleci.com/pipelines/github/pytorch/vision/7858/workflows/3f6aa4f8-ba63-48d6-b939-4f97af2e3573/jobs/555947 for one example

--- a/.circleci/unittest/linux/scripts/environment.yml
+++ b/.circleci/unittest/linux/scripts/environment.yml
@@ -9,7 +9,8 @@ dependencies:
   - codecov
   - pip
   - libpng
-  - jpeg
+  # NOTE: Pinned to fix issues with size_t on Windows
+  - jpeg <=9b
   - ca-certificates
   - pip:
     - future

--- a/.circleci/unittest/windows/scripts/environment.yml
+++ b/.circleci/unittest/windows/scripts/environment.yml
@@ -9,7 +9,8 @@ dependencies:
   - codecov
   - pip
   - libpng
-  - jpeg
+  # NOTE: Pinned to fix issues with size_t on Windows
+  - jpeg <=9b
   - ca-certificates
   - pip:
     - future

--- a/packaging/torchvision/meta.yaml
+++ b/packaging/torchvision/meta.yaml
@@ -9,7 +9,8 @@ requirements:
   build:
     - {{ compiler('c') }} # [win]
     - libpng
-    - jpeg
+    # NOTE: Pinned to fix issues with size_t on Windows
+    - jpeg <=9b
     # NOTE: The only ffmpeg version that we build is actually 4.2
     - ffmpeg >=4.2  # [not win]
 
@@ -24,7 +25,8 @@ requirements:
     - python
     - libpng
     - ffmpeg >=4.2  # [not win]
-    - jpeg
+    # NOTE: Pinned to fix issues with size_t on Windows
+    - jpeg <=9b
     - pillow >=5.3.0
     {{ environ.get('CONDA_PYTORCH_CONSTRAINT') }}
     {{ environ.get('CONDA_CUDATOOLKIT_CONSTRAINT') }}
@@ -51,7 +53,8 @@ test:
     - pytest
     - scipy
     - av >=8.0.1
-    - jpeg
+    # NOTE: Pinned to fix issues with size_t on Windows
+    - jpeg <=9b
     - ca-certificates
 
 

